### ISSUE TITLE
The package does not handle UTF-8 prompts properly

### ISSUE THREAD

**Issue by Ailrun** [fxfactorial/ocaml-linenoise#13]
# Description
After I used `λ ` as a prompt message, `ocaml-linenoise` displayed a cursor at a wierd position.
For example, when I typed `abc` (without a space), my terminal displays something like this.
```
λ abc _
     ^ Here is a weird space
```

# What you expected
`ocaml-linenoise` handles `λ` (and any other UTF-8 encoded characters) in a proper way so that it places characters and the cursor at expected positions like the following.
```
λ abc_
     ^ This underscore is the cursor.
```

# Reproduction
https://github.com/Ailrun/ocaml-linenoise-issue-13
This repository also includes instructions to reproduce the bug.

**Comment by c-cube** [comment]
Indeed, the C linenoise library doesn't support unicode. I don't see a simple way to fix that for now, but it should definitely be on the road map.
